### ISSUE TITLE
Uncaught TypeError: Cannot read property 'focus' of null

### ISSUE THREAD

**Issue by CamiloSerrano** [jeremyramin/terminal-plus#277]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.7.3
**System**: Ubuntu 16.04
**Thrown From**: [terminal-plus](https://github.com/jeremyramin/terminal-plus) package, v0.14.5

### Stack Trace

Uncaught TypeError: Cannot read property 'focus' of null

```
At /home/cserrano/.atom/packages/terminal-plus/lib/status-bar.coffee:92

TypeError: Cannot read property 'focus' of null
    at /home/cserrano/.atom/packages/terminal-plus/lib/status-bar.coffee:92:23
```

### Commands

```
  5x -1:41.5.0 core:undo (atom-text-editor.editor.is-focused)
     -1:37.2.0 core:save (atom-text-editor.editor.is-focused.autocomplete-active)
     -1:35.2.0 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
     -1:34.7.0 editor:select-to-end-of-line (atom-text-editor.editor.is-focused)
     -1:34.1.0 core:select-down (atom-text-editor.editor.is-focused)
     -1:33.1.0 core:move-left (atom-text-editor.editor.is-focused)
     -1:32.4.0 core:move-down (atom-text-editor.editor.is-focused)
     -1:32 editor:move-to-first-character-of-line (atom-text-editor.editor.is-focused)
     -1:31.8.0 snippets:next-tab-stop (atom-text-editor.editor.is-focused)
     -1:31.8.0 snippets:expand (atom-text-editor.editor.is-focused)
     -1:31.8.0 editor:indent (atom-text-editor.editor.is-focused)
     -1:31 core:save (atom-text-editor.editor.is-focused)
     -1:12.5.0 core:copy (atom-text-editor.editor.is-focused)
     -1:10.2.0 core:paste (atom-text-editor.editor.is-focused)
     -1:09.7.0 core:save (atom-text-editor.editor.is-focused)
     -0:02.4.0 core:cancel (atom-workspace.workspace.scrollbars-visible-always.theme-atom-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "themes": [
      "one-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "terminal-plus": {
    "core": {},
    "style": {},
    "toggles": {
      "autoClose": true
    }
  }
}
```

### Installed Packages

```coffee
# User
project-view, v0.5.0
python-tools, v0.6.7
terminal-plus, v0.14.5
tree-view-git-status, v0.2.3

# Dev
No dev packages
```

**Comment by kachkaev** [comment]
I also have this issue from time to time in the latest Atom. Hard to tell what precedes it – the notification fires up randomly.

**Comment by CamiloSerrano** [comment]
Is at the moment in which it is in the editor and realizes tab apparently does not focus in the editor or in the terminal

**Comment by kksevta** [comment]
I am using Ubuntu 14.04 .
It is occurring every time if currently I am in edit mode of terminal window and then directly click top icon in launcher at left

**Comment by KMayne** [comment]
I get this issue when I tap Alt (opens the HUD) on Ubuntu 16.04

**Comment by BusfyJan** [comment]
I get this issue when I focus terminal and then open ubuntu (14.04) launcher using windows key or by clicking on launcher icon.

**Comment by sangeeth96** [comment]
The error seems to occur when the terminal is in focus and I press the <kbd>alt</kbd> (HUD) or the <kbd>win</kbd> key (Dash) or even by clicking on the launcher icon as @BusfyJan mentioned above.

**Comment by 3DEsprit** [comment]
Ubuntu 16.04
Atom 1.8.0

I am also getting this issue when after moving focus from Atom. At least that is the only time I've noticed it, switching to Chrome or Postman; I have not tried to reproduce it.

**Comment by ryanrahman26** [comment]
I'm on Ubuntu 16.04 LTS

I go this when i left my cursor on terminal to switch firefox and switch agan to Atom then tadaaa this error show

Hope this issue can fixed

**Comment by WaKeMaTTa** [comment]
- Ubuntu 16.04 LTS
- Terminal Plus v0.14.5

Still exist the bug. :bug:

**Comment by 3DEsprit** [comment]
Lots of bugs still exist, including a breaking bug with newer versions of Atom. It seems that the project has been abandoned. Switch to Terminal Fusion, and you'll feel at home, without bugs. :)

**Comment by WaKeMaTTa** [comment]
Thanks @3DEsprit i will give a try to [Terminal Fusion](https://atom.io/packages/terminal-fusion).

**Comment by kachkaev** [comment]
Thanks for pointing out that this repo is abandoned @3DEsprit! There are quite a few other alternatives actually, here are some:
http://atom-packages.directory/category/tools-terminal/

[`atom-ide-terminal`](https://github.com/platformio/platformio-atom-ide-terminal) seems to be the second most popular these days (after `terminal-plus`); it is actually a fork of this repo and the origin for Terminal Fusion.

**Comment by 3DEsprit** [comment]
Yes, there are quite a few as mentioned in another issue, but I had to pass in TF, as it seems to be the most involved pty.js project so far. :)
